elyra/pipeline/tests/test_validation.py

```diff
@@ -0,0 +1,369 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+
+import pytest
+
+from elyra.pipeline.validate import PipelineValidationManager
+from elyra.pipeline.validate import ValidationResponse
+
+
+@pytest.fixture
+def load_pipeline():
+    def _function(pipeline_filepath):
+        response = ValidationResponse()
+
+        with open(f'elyra/pipeline/tests/resources/validation_pipelines/{pipeline_filepath}') as f:
```

Comment:
This (and the `validation_manager` fixture below) introduces the requirement that testing must occur from the root of the repo.  This will break normal CLI and IDE functionality.

We should be using temporary directories for all tests and emulate installation/deployment relative to those directories.  However, that will take some time to implement and these tests will work fine in fixed environments used in CI - so can be deferred IMO.